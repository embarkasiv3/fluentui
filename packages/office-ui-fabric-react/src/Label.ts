export * from './packages/react-fundamentals/Label';
