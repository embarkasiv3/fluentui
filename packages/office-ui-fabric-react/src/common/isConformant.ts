import { isConformant as isConformantBase, IsConformantOptions } from '@fluentui/react-conformance';

export function isConformant(testInfo: IsConformantOptions) {
  const defaultOptions = {
    disabledTests: ['has-docblock'],
  };

  isConformantBase(defaultOptions, testInfo);
}
