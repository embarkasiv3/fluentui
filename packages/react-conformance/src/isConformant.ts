import { TestObject, TestingOptions } from './types';
import { withCustomConfig } from 'react-docgen-typescript';
import * as path from 'path';
import { defaultTests } from './defaultTests';

export function isConformant(testInfo: TestingOptions, extraTests?: TestObject) {
  const { componentPath, displayName } = testInfo;
  const tsconfigPath = path.join(process.cwd(), 'tsconfig.json');
  const parser = withCustomConfig(tsconfigPath, {});
  const components = parser.parse(componentPath);
  const mainComponents = components.filter(comp => comp.displayName === displayName);

  if (mainComponents.length === 1) {
    const componentInfo = mainComponents[0];

    for (const test of Object.keys(defaultTests)) {
      defaultTests[test](componentInfo, testInfo);
    }
    if (extraTests) {
      for (const test of Object.keys(extraTests)) {
        extraTests[test](componentInfo, testInfo);
      }
    }
  } else {
    if (components.length === 0) {
      throw new Error('No exported components in path: ' + componentPath);
    } else {
      throw new Error(
        `No component with name '${displayName}' was found at ${componentPath}. ` +
          `These are the exported component names: ${components.map(component => component.displayName).join(', ')}`,
      );
    }
  }
}
