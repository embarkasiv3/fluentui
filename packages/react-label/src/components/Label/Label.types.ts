import * as React from 'react';
import { ComponentProps, ComponentState } from '@fluentui/react-utilities';

/**
 * Label Props
 */
export interface LabelProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
  /*
   * TODO Add props and slots here
   * Any slot property should be listed in the labelShorthandProps array below
   * Any property that has a default value should be listed in LabelDefaultedProps as e.g. 'size' | 'icon'
   */
}

/**
 * Names of the shorthand properties in LabelProps
 */
export const labelShorthandProps = [] as const; // TODO add shorthand property names

/**
 * Names of the shorthand properties in LabelProps
 */
export type LabelShorthandProps = typeof labelShorthandProps[number];

/**
 * Names of LabelProps that have a default value in useLabel
 */
export type LabelDefaultedProps = never; // TODO add names of properties with default values

/**
 * Label State
 */
export type LabelState = ComponentState<React.Ref<HTMLElement>, LabelProps, LabelShorthandProps, LabelDefaultedProps>;
