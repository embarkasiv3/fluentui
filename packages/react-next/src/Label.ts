import './version';
export * from '@fluentui/react/lib/Label';
