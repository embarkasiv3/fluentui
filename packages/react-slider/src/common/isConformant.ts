import { isConformant as baseIsConformant, IsConformantOptions } from '@fluentui/react-conformance';

export function isConformant(testInfo: IsConformantOptions) {
  const defaultOptions = {
    disabledTests: ['has-docblock', 'kebab-aria-attributes'],
  };

  baseIsConformant(defaultOptions, testInfo);
}
