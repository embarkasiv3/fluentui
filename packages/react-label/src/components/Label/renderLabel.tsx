import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import { labelShorthandProps, LabelState } from './Label.types';

/**
 * Render the final JSX of Label
 */
export const renderLabel = (state: LabelState) => {
  const { slots, slotProps } = getSlots(state, labelShorthandProps);

  return (
    <slots.root {...slotProps.root}>
      {/* TODO Add additional slots in the appropriate place */}
      {state.children}
    </slots.root>
  );
};
