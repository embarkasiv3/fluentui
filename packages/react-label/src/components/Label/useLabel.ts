import * as React from 'react';
import { makeMergeProps, resolveShorthandProps, useMergedRefs } from '@fluentui/react-utilities';
import { LabelProps, labelShorthandProps, LabelState } from './Label.types';

const mergeProps = makeMergeProps<LabelState>({ deepMerge: labelShorthandProps });

/**
 * Create the state required to render Label.
 *
 * The returned state can be modified with hooks such as useLabelStyles,
 * before being passed to renderLabel.
 *
 * @param props - props from this instance of Label
 * @param ref - reference to root HTMLElement of Label
 * @param defaultProps - (optional) default prop values provided by the implementing type
 */
export const useLabel = (props: LabelProps, ref: React.Ref<HTMLElement>, defaultProps?: LabelProps): LabelState => {
  const state = mergeProps(
    {
      ref: useMergedRefs(ref, React.useRef(null)),
    },
    defaultProps && resolveShorthandProps(defaultProps, labelShorthandProps),
    resolveShorthandProps(props, labelShorthandProps),
  );

  return state;
};
