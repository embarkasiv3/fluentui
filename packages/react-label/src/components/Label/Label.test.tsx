import * as React from 'react';
import { Label } from './Label';
import * as renderer from 'react-test-renderer';
import { ReactWrapper } from 'enzyme';
import { isConformant } from '../../common/isConformant';

describe('Label', () => {
  isConformant({
    Component: Label,
    displayName: 'Label',
  });

  let wrapper: ReactWrapper | undefined;

  afterEach(() => {
    if (wrapper) {
      wrapper.unmount();
      wrapper = undefined;
    }
  });

  /**
   * Note: see more visual regression tests for Label in /apps/vr-tests.
   */
  it('renders a default state', () => {
    const component = renderer.create(<Label>Default Label</Label>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
