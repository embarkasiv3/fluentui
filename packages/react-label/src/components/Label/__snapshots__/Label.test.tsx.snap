// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders a default state 1`] = `
<div
  className="__9et6fu0 "
>
  Default Label
</div>
`;
